packages/core/src/browser/fetchObservable.ts

```diff
@@ -96,21 +97,31 @@ function afterSend(
   responsePromise: Promise<Response>,
   startContext: FetchStartContext
 ) {
-  const reportFetch = (response: Response | Error) => {
-    const context = startContext as unknown as FetchResolveContext
+  const context = startContext as unknown as FetchResolveContext
+
+  function reportFetch(partialContext: Partial<FetchResolveContext>) {
     context.state = 'resolve'
-    if ('stack' in response || response instanceof Error) {
-      context.status = 0
-      context.isAborted = response instanceof DOMException && response.code === DOMException.ABORT_ERR
-      context.error = response
-    } else if ('status' in response) {
-      context.response = response
-      context.responseType = response.type
-      context.status = response.status
-      context.isAborted = false
-    }
+    assign(context, partialContext)
     observable.notify(context)
   }
 
-  responsePromise.then(monitor(reportFetch), monitor(reportFetch))
+  responsePromise.then(
+    monitor((response) => {
+      reportFetch({
+        response,
+        responseType: response.type,
+        status: response.status,
+        isAborted: false,
+      })
+    }),
+    monitor((error: Error) => {
+      const isAborted =
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏You could inline it
```ts
    reportFetch({
        status: 0,
        isAborted:
          context.init?.signal?.aborted || (error instanceof DOMException && error.code === DOMException.ABORT_ERR),
        error,
      })
```

Reply:
Done